### DISCUSSION TITLE
Monitoring & Logging for Teleport

### DISCUSSION THREAD

**Discussion by PallaviPalisetti** [gravitational/teleport#15002] [Q&A]
Hi there,

what are the monitoring and logging options available for teleport. if  Teleport is installed on virtual server.

Thanks

**Comment by scottpgallagher** [comment]
Can you explain a little more what your trying to achieve so we can better answer your question?

**Reply by PallaviPalisetti** [reply]
thanks for quick response.

For monitoring, we can pick this as an example, that there are lots of these slow queries from SQLite, and we want to monitor these, and want to know more why that's happening, as we're using the highest IOPS available Disk.
![image](https://user-images.githubusercontent.com/87689454/181587785-d77611c3-6915-48f5-919b-347ad2ec859e.png)

**Comment by scottpgallagher** [comment]
For monitoring/alerting you can send Teleport logs off to a logging or SIEM tool, but addressing the error itself is there anything else special about these servers, anything like scanners, intrusion detect/prevention? stuff like that. On another product we did see lots of issues with security software returning unexpected results to syscalls, although this is more like stdout/stderr is breaking. Is anything else going on the host around the same time? Is journald dieing or restarting or something? Anything in dmesg about kernel panics or errors?
